docker/linux/Dockerfile

```diff
@@ -0,0 +1,73 @@
+ARG IMAGE_OS_BASE=almalinux
+ARG IMAGE_OS_VERSION=8
+FROM ${IMAGE_OS_BASE}:${IMAGE_OS_VERSION}
+
+# Setting up base OS with GCC 11 and Python 3.12
+RUN set -e \
+    && rpm --import https://repo.almalinux.org/almalinux/RPM-GPG-KEY-AlmaLinux \
+    && dnf -y update \
+    && dnf install -y dnf-plugins-core https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm \
+    && dnf config-manager --set-enabled powertools \
+    && INSTALL_PKGS="git unzip bzip2 gcc-toolset-11 gcc-toolset-11-gcc gcc-toolset-11-gcc-c++ gcc-toolset-11-gdb redhat-lsb cmake make ninja-build python3.12 python3.12-pip" \
+    && dnf install -y --setopt=tsflags=nodocs $INSTALL_PKGS \
+    && rpm -V $INSTALL_PKGS \
+    && dnf clean all \
+    && rm -rf /var/cache/dnf
+
+# Adding 'default' user to do the work
+ENV HOME=/tmp/work
+COPY docker/linux/docker-entrypoint.sh /usr/bin/container-entrypoint.sh
+RUN set -e \
+    && mkdir -p ${HOME} \
+    && groupadd -r default -f -g 1001 \
+    && useradd -r default -g default -d ${HOME} -s /sbin/nologin -c "Default Application User" \
+    && chown -R default:default ${HOME} \
+    && chmod -R 755 ${HOME} \
+    && chown default:default /usr/bin/container-entrypoint.sh \
+    && chmod 555 /usr/bin/container-entrypoint.sh
+WORKDIR ${HOME}
+
+# Putting Python user base outside of the workdir so we can mount it from the outside
+ENV PYTHONUSERBASE=/opt/pyuser
+RUN set -e \
+    && mkdir -p ${PYTHONUSERBASE} \
+    && chown -R default:default ${PYTHONUSERBASE} \
+    && chmod -R 755 ${PYTHONUSERBASE}
+ENV PATH=${PYTHONUSERBASE}/bin:${PATH}
+
+# Putting conan cache outside of the workdir due to mounting and slow chmod/chown in docker volumes on RHEL8 kernels
+# (the cache can get big with lots of files). Using RO mode to support consuming the cache with different UID/GID.
+ENV CONAN_CACHE_NO_LOCKS=1
+ENV CONAN_READ_ONLY_CACHE=1
+ENV CONAN_USER_HOME=/opt/conan
+RUN set -e \
+    && mkdir -p ${CONAN_USER_HOME} \
+    && chown -R default:default ${CONAN_USER_HOME} \
+    && chmod -R 755 ${CONAN_USER_HOME}
+
+# Setting up the Python environment
+USER default
+RUN python3.12 -m pip install --user --no-cache-dir --upgrade pip wheel "conan<2.0"
+
+# Pre-installing the Conan cache
+ENV CONAN_RECIPES=/tmp/conan-recipes
+COPY --chown=default ./conan ${CONAN_RECIPES}
+
+ARG CESDK_VERSION=3.2.10650
+RUN conan create -pr ${CONAN_RECIPES}/profiles/linux-gcc112 ${CONAN_RECIPES}/cesdk cesdk/${CESDK_VERSION}@esri-rd-zurich/stable
+
+ARG HDK_HOST_LOCATION=/opt/houdini
+ARG HDK_VERSION=20.0.751
+COPY --from=houdini --chown=default . /tmp/Houdini${HDK_VERSION}
+RUN conan create -pr ${CONAN_RECIPES}/profiles/linux-gcc112 -e HOUDINI_INSTALL=/tmp/Houdini${HDK_VERSION} ${CONAN_RECIPES}/houdini houdini/${HDK_VERSION}@sidefx/stable
```

Comment:
You will need two bind-mounts here. In case that any of the bind-mounted dirs need to be r/w, the full set of options are documented in the [Dockerfile reference](https://docs.docker.com/reference/dockerfile/#run---mount).

Reply:
done